common/src/main/java/discord4j/common/JacksonResourceProvider.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discord4j.common;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import discord4j.common.jackson.PossibleModule;
+import discord4j.common.jackson.UnknownPropertyHandler;
+
+import java.util.function.Function;
+
+/**
+ * Provides a centralized Jackson 2.9 {@link com.fasterxml.jackson.databind.ObjectMapper} allowing customization and
+ * reuse across the application.
+ */
+public class JacksonResourceProvider {
+
+    private static final Function<ObjectMapper, ObjectMapper> initializer = mapper -> mapper
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
+            .registerModules(new PossibleModule(), new Jdk8Module());
+
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Create a default {@link com.fasterxml.jackson.databind.ObjectMapper} that allows any field visibility,
+     * registers modules to handle Discord4J specific mappings and ignores unknown properties.
+     */
+    public JacksonResourceProvider() {
```

Comment:
Can this call the overloaded constructor with the used `mapper`?

Reply:
Sure, looks cleaner.